### PR TITLE
Fixes to run on python 3.10

### PR DESCRIPTION
**Pull request by thecharlieblake**

### CODE REVIEW

File: unit_scaling/utils.py [hunk 1]
```diff
@@ -222,7 +222,7 @@ def trace(
         """Adds the `target_to_function` dict to the graph so the interpreter can use it
         downstream."""
         graph = super().trace(root, concrete_args)
-        if graph._tracer_extras is None:
+        if not hasattr(graph, "_tracer_extras") or graph._tracer_extras is None:
```
Comment (DouglasOrr) [inline unit_scaling/utils.py:225]: If I understand the test (precendence of `or` and `not` always gets me), could this use `getattr(graph, "_tracer_extras", None) is None`?

Comment (DouglasOrr) [inline unit_scaling/utils.py:225]: If I understand the test (precendence of `or` and `not` always gets me), could this use `getattr(graph, "_tracer_extras", None) is None`?

**Review by DouglasOrr — COMMENTED** [review]
Optional comment. On the title, is this for PyTorch 1.10?

**Review by DouglasOrr — APPROVED** [review]

File: unit_scaling/utils.py (hunk 1 above)
Comment (thecharlieblake) [reply unit_scaling/utils.py:225]: My vscode gives me warnings locally if I don't use `hasattr` (no actual issues, but red lines are annoying) so tempted to keep it this way for now...

**Pull request closed without merging**